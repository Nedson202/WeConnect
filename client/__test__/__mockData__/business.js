export default {
  business: {
    id: 1,
    name: 'alpha tech',
    email: 'altech@caria.net',
    address: '2r close sau paulo',
    location: 'sau paulo',
    categories: 'tech',
    description: 'this is a simple meal made with rice and yam',
    image: 'image/connect.jpg'
  },
  allBusiness: [
    {
      id: 1,
      name: 'alpha tech',
      email: 'altech@caria.net',
      address: '2r close sau paulo',
      location: 'sau paulo',
      categories: 'tech',
      description: 'this is a simple meal made with rice and yam',
      image: 'image/connect.jpg',
      userId: 1
    },
    {
      id: 2,
      name: 'techudia',
      email: 'testech@caria.net',
      address: '2r close sau paulo',
      location: 'sau paulo',
      categories: 'tech',
      description: 'this is a simple meal made with rice and yam',
      image: 'image/connect.jpg',
      userId: 1
    },
    {
      id: 3,
      name: 'Lancorp',
      email: 'lancorp@biz.net',
      address: '2r close sau paulo',
      location: 'sau paulo',
      categories: 'tech',
      description: 'this is a simple meal made with rice and yam',
      image: 'image/connect.jpg',
      userId: 3
    }
  ],
  usersBusiness: [
    {
      id: 1,
      name: 'alpha tech',
      email: 'altech@caria.net',
      address: '2r close sau paulo',
      location: 'sau paulo',
      categories: 'tech',
      description: 'this is a simple meal made with rice and yam',
      image: 'image/connect.jpg',
      userId: 1
    },
    {
      id: 2,
      name: 'techudia',
      email: 'testech@caria.net',
      address: '2r close sau paulo',
      location: 'sau paulo',
      categories: 'tech',
      description: 'this is a simple meal made with rice and yam',
      image: 'image/connect.jpg',
      userId: 1
    }
  ],
  locations: [
    {
      id: 1,
      location: 'lao'
    },
    {
      id: 2,
      location: 'hawaii'
    }
  ],
  categories: [
    {
      id: 1,
      category: 'finance'
    },
    {
      id: 2,
      category: 'tech'
    }
  ],
  paginationResult: {
    page: 1,
    limit: 6,
    totalBusiness: 10
  },
  review: {
    message: 'Great business',
    rating: 3
  },
  loadingStatus: {
    isLoading: false
  }
};
