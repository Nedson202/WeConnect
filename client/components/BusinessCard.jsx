import React from 'react';
import Truncate from 'react-truncate';
import { Link } from 'react-router-dom';
import green from '../images/default.jpeg';

/**
 * @description react stateless component for card template
 * 
 * @return {Object} action dispatched by the action creator
 */
const BusinessCard =({ business }) => {
  return (
    <div className="col col-sm-10 col-md-6 col-lg-4 card-style">
      <div className="card business-card">
        <div className="zoom"> 
          <Link to={`profile/${business.id}`}>
            <img className="card-img-top image-fluid" src={green} alt="business" height="200px" />
          </Link>
        </div>
        <div className="card-body">
          <p className="mb-0 small font-weight-medium text-uppercase mb-1 text-muted lts-2px">
            {business.category}
          </p>
          <h5 className="card-title text-center">{business.name}</h5>
          <Truncate 
            line={2}
          >
            {business.description}            
          </Truncate>
          <div className="profile-icon-style">
            <p className="card-icon"><i className="fa fa-map-marker style" /><span>{ business.location }</span></p>          
          </div>
          <div>
            <Link to={`profile/${business.id}`}>Read more</Link>
          </div>
        </div>
      </div>
    </div>
  );
}

export default BusinessCard;