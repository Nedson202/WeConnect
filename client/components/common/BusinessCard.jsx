import React from 'react';
import Truncate from 'react-truncate';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import green from '../../images/default.jpeg';

/**
 * @description react stateless component for card template
 *
 * @return {Object} action dispatched by the action creator
 */
const BusinessCard = ({ business }) => {
  const {
    name, id, image, category, location, description
  } = business;
  return (
    <div className="col-sm-12 col-md-6 col-lg-4 card-style">
      <div className="card business-card">
        <div className="card-header text-center textcase">
          <h5>{name}</h5>
        </div>
        <div className="zoom">
          <Link to={`profile/${id}`}>
            <img className="card-img-top image-fluid" src={image || green} alt="business" height="200px" />
          </Link>
        </div>
        <hr style={{ marginTop: 0, marginBottom: 0 }} />
        <div className="card-body">
          <p className="mb-0 small text-uppercase mb-1 lts-2px pull-left">
            {category}
          </p>
          <p className="mb-0 small text-uppercase mb-1 lts-2px pull-right">
            { location }
          </p>
          <div className="card-title text-center textcase"><p /></div>
          <Truncate
            line={2}
          >
            {description}
          </Truncate>
          <div>
            <Link to={`profile/${id}`}>
              <button className="btn btn-outline-success" id="read-more">Read more</button>
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

BusinessCard.propTypes = {
  business: PropTypes.object.isRequired,
};

export default BusinessCard;
