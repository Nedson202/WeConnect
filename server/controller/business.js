import businesses from '../model/business';
/**
  *
  *Business class to handle actions like registration, business filtering
  *updating business, retrieving business, deletion of business
  *@class
  *
*/
class Businesses {
  /**
    *
    *Register business
    *@param {any} req - request value - handles data coming from the user
    *@param {any} res - response value - this is the response gotten after
    interaction with the Api routes
    *@return {json} response object gotten
    *@memberof Businesses
  */
  static createBusiness(req, res) {
    const {
      name, email, address, location, category
    } = req.body;

    const filteredBusiness = businesses.filter(business => business.name === name)[0];

    if (filteredBusiness) {
      res.status(400).json({
        message: 'Business with name, is already taken',
        error: true
      });
    }

    if (!filteredBusiness) {
      businesses.push({
        id: businesses[businesses.length - 1].id + 1,
        name,
        email,
        address,
        location,
        category
      });
    }

    return res.status(201).json({
      message: 'Business registration successful',
      error: 'false'
    });
  }
}

export default Businesses;
