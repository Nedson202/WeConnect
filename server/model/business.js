const businesses = [
  {
    id: 1,
    name: 'alans corp',
    email: 'alanscorpy@alc.com',
    address: '223rd road ondo',
    location: 'ondo',
    category: 'oil',
    reviews: [
      {
        reviewer: 'michael',
        messsage: 'great business'
      }
    ]
  },
  {
    id: 2,
    name: 'alans',
    email: 'oppson@oppo.com',
    address: '12 cl allen avenue',
    location: 'enugu',
    category: 'mobile',
    reviews: [
      {
        reviewer: 'michael',
        messsage: 'great business'
      }
    ]
  }
];

export default businesses;
