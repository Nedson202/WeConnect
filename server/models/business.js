
module.exports = (sequelize, DataTypes) => {
  const Business = sequelize.define('Business', {
    name: {
      type: DataTypes.STRING,
      required: true,
      unique: {
        args: true,
        msg: 'Business with name is already registered'
      }
    },
    email: {
      type: DataTypes.STRING,
      required: true,
      unique: {
        args: true,
        msg: 'Email is already taken'
      }
    },
    address: {
      type: DataTypes.STRING,
      required: true
    },
    location: {
      type: DataTypes.STRING,
      required: true
    },
    category: {
      type: DataTypes.STRING,
      required: true
    }
  });

  Business.associate = (models) => {
    Business.belongsTo(models.Category, {
      foreignKey: 'categoryId',
      onDelete: 'cascade',
      hooks: true
    });
    Business.belongsTo(models.User, {
      foreignKey: 'userId',
      onDelete: 'cascade',
      hooks: true
    });
    Business.hasMany(models.Review, {
      foreignKey: 'businessId',
      as: 'businessReviews',
      onDelete: 'cascade',
      hooks: true
    });
  };

  return Business;
};
