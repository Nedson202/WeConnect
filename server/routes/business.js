import Auth from '../controller/user';
import Businesses from '../controller/business';
import Reviews from '../controller/review';
import validator from '../middlewares/validator';
import sorter from '../middlewares/business-filterer';
import errorHandler from '../middlewares/error-handler';

export default (route) => {
  route.post('/api/v1/businesses', validator.registerBusiness, errorHandler, Businesses.createBusiness);
  route.get('/api/v1/businesses', sorter.sortQuery, Businesses.getBusiness);
  route.get('/api/v1/businesses/:businessId', Businesses.getOneBusiness);
  route.put('/api/v1/businesses/:businessId', sorter.filterBusiness, Businesses.updateBusiness);
  route.delete('/api/v1/businesses/:businessId', sorter.filterBusiness, Businesses.deleteBusiness);
  route.post('/api/v1/businesses/:businessId/reviews/', validator.reviews, errorHandler, Reviews.postReview);
  route.get('/api/v1/businesses/:businessId/reviews/', Reviews.getReview);
};
