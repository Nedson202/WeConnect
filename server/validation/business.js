import Validator from 'validator';
import isEmpty from 'lodash/isEmpty';

export default function businessValidator(data) {
  const errors = {};

  if (Validator.isEmpty(data.name)) {
    errors.name = 'This field is required';
  }
  if (Validator.isEmpty(data.email)) {
    errors.email = 'This field is required';
  }
  if (!Validator.isEmail(data.email)) {
    errors.email = 'Email is invalid';
  }
  if (Validator.isEmpty(data.address)) {
    errors.address = 'This field is required';
  }
  if (Validator.isEmpty(data.location)) {
    errors.location = 'This field is required';
  }
  if (Validator.isEmpty(data.category)) {
    errors.category = 'This field is required';
  }

  return {
    errors,
    isValid: isEmpty(errors)
  }
}
