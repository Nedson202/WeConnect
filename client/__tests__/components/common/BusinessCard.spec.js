import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import BusinessCard from '../../../components/common/BusinessCard';
import business from '../../__mockData__/business';

configure({ adapter: new Adapter() });

let props;
const setup = () => {
  props = business;

  return shallow(<BusinessCard {...props} />);
};


describe('Component: BusinessCard', () => {
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(7);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('Link').length).toBe(2);
    expect(wrapper.find('Truncate').length).toBe(1);
    expect(wrapper.find('img').length).toBe(1);
    expect(wrapper.length).toBe(1);
  }) 
});