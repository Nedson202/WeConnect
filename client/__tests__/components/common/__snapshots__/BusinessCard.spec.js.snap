// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BusinessCard should render 1`] = `
<div
  className="col-sm-12 col-md-6 col-lg-4 card-style"
>
  <div
    className="card business-card"
  >
    <div
      className="card-header text-center textcase"
    >
      <h5>
        alpha tech
      </h5>
    </div>
    <div
      className="zoom"
    >
      <Link
        replace={false}
        to="profile/1"
      >
        <img
          alt="business"
          className="card-img-top image-fluid"
          height="200px"
          src="image/connect.jpg"
        />
      </Link>
    </div>
    <hr
      style={
        Object {
          "marginBottom": 0,
          "marginTop": 0,
        }
      }
    />
    <div
      className="card-body"
    >
      <p
        className="mb-0 small text-uppercase mb-1 lts-2px pull-left"
      />
      <p
        className="mb-0 small text-uppercase mb-1 lts-2px pull-right"
      >
        sau paulo
      </p>
      <div
        className="card-title text-center textcase"
      >
        <p />
      </div>
      <Truncate
        ellipsis="â€¦"
        line={2}
        lines={1}
        trimWhitespace={false}
      >
        this is a simple meal made with rice and yam
      </Truncate>
      <div>
        <Link
          replace={false}
          to="profile/1"
        >
          <button
            className="btn btn-outline-success"
            id="read-more"
          >
            Read more
          </button>
        </Link>
      </div>
    </div>
  </div>
</div>
`;
